### ISSUE TITLE
Feature Request: Inline display of previously executed lines

### ISSUE THREAD

**Issue by yingzhu146** [michaelb/sniprun#61]
**Is your feature request related to a problem? Please describe.**
My working memory is limited. Instead of remembering runtime variables and previous results I love to seem them. There's many approaches to solve this, like variable viewers used in debuggers or available as jupyter plugins, but one of my favorite ones are [Atom's Hydrogen](https://atom.io/packages/hydrogen) and inline result display a la [codi](https://github.com/metakirby5/codi.vim) as both are highly space efficient. This idea came out of #54 .

**Describe the solution you'd like**

Now what would be beyond amazing is if we could display the result of previously executed lines like [nvim-hlslense](https://github.com/kevinhwang91/nvim-hlslens) displays the grey search counts here

<img width="793" alt="CleanShot 2021-04-19 at 22 28 09@2x" src="https://user-images.githubusercontent.com/59233975/115328434-9f819700-a15e-11eb-8ec0-6e4746b11e2d.png">

**Describe alternatives you've considered**
- use a debugger with variable explorer
- use jupyter notebook/vscode's variable explorer

**Comment by michaelb** [comment]
alright, yet again a good idea !

The exact term you're searching for is 'virtual text', and while a bit new, there should be nothing too complicated (i hope) with implementing such a display.

I'll get working on that, though I can't promise results for tomorrow, since i've my own job, plus other features that require some (most) of my time.

**Comment by yingzhu146** [comment]
No immediate rush, again thanks a ton and obviously most happy to test anything!

**Comment by michaelb** [comment]
There is some working stuff in the dev branch!

Plug 'michaelb/sniprun', {'do': 'bash install.sh 1', 'branch':'dev'}

Needless to say, I'll have to sanitize a bit the default behavior,and setup options etc... but that's work for another day, it's already late.

Not that I absolutely need it, but feedback is always welcome, especially regarding this point: what would you expect to see when a multiples lines are printed? (virtual text can only be set on one line, unlike hydrogen stuff, though that may be workaroundable a the cost of some compromises)

**Comment by yingzhu146** [comment]
I can't believe you just built this in less than a day - it's fantastic, and works super well! thanks a lot.

I think a cool default behavior would be: show a virtual window with full length output for most recent executed, and show shortened virtual text for rest.  (I actually don't know if it's possible to show multiple floating windows but that would be potentially cool too then it's more a la atom)

I do run into [this](https://github.com/michaelb/sniprun/issues/47#issuecomment-823434685) still though so I've done limited testing.

Again, your work is fantastic so thanks a ton!

**Comment by michaelb** [comment]
actually, the problem you linked is expected behavior (see comment) and it's related to #41 

Floating windows are a very nice idea I didn't have, again, thanks!

**Comment by michaelb** [comment]
This has been closed a bit... automatically, but I hope you get this notification @yingzhu146 

Your feature request has been merged into sniprun v0.5.3 along with other things, I hope you'll find the solution satisfying, if not don't hesitate to reopen an issue!

Regards, michael

**Comment by yingzhu146** [comment]
Hi @michaelb  - this is absolutely outstanding. excited to try it out more. More generally though, I found the way you helped me and others and showed open-ness towards my request & implemented it in less than 12 hours very inspiring. I hope others find this exchange just as inspiring as me - I've created a (post on reddit)[https://www.reddit.com/r/neovim/comments/mxljax/a_case_of_awesomeness_in_the_neovim_plugin/] to highlight both the feature and this exchange, I hope you don't mind, and again thanks a ton!
